spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -420,12 +421,14 @@ private Scan buildBatchScan() {
   private Scan buildBatchScan(Long snapshotId, Long asOfTimestamp, String branch, String tag) {
     Schema expectedSchema = schemaWithMetadataColumns();
 
+    SparkMetricsReporter reporter = new SparkMetricsReporter();
```

Comment:
I don't think there is any value in this class at this point. Let's remove Spark classes from this PR and add them when we know how the reporting logic will look like.